src/Tools/babylon.tools.ts

```diff
@@ -814,6 +814,12 @@
                 if (doNotCopyList && doNotCopyList.indexOf(prop) !== -1) {
                     continue;
                 }
+
+                let descriptor = Object.getOwnPropertyDescriptor(source, prop);
+                //descriptor should exist, should be writable, and should not be a getter.

```

Comment:
Will this prevent copying simple field members?

Reply:
no, getOwnPropertyDescriptor returns a descriptor for each class member that is not a getter. These are the descriptors available on a regular mesh:

```
[
  "state",
  "metadata",
  "doNotSerialize",
  "_isDisposed",
  "animations",
  "_ranges",
  "_isEnabled",
  "_isReady",
  "_currentRenderId",
  "_parentRenderId",
  "_animationPropertiesOverride",
  "onDisposeObservable",
  "_behaviors",
  "name",
  "id",
  "_scene",
  "uniqueId",
  "_cache",
  "_rotation",
  "_scaling",
  "_isDirty",
  "billboardMode",
  "scalingDeterminant",
  "infiniteDistance",
  "position",
  "_localWorld",
  "_worldMatrix",
  "_worldMatrixDeterminant",
  "_absolutePosition",
  "_pivotMatrix",
  "_postMultiplyPivotMatrix",
  "_isWorldMatrixFrozen",
  "onAfterWorldMatrixUpdateObservable",
  "_nonUniformScaling",
  "_facetNb",
  "_partitioningSubdivisions",
  "_partitioningBBoxRatio",
  "_facetDataEnabled",
  "_facetParameters",
  "_bbSize",
  "_subDiv",
  "_facetDepthSort",
  "_facetDepthSortEnabled",
  "onCollideObservable",
  "onCollisionPositionChangeObservable",
  "onMaterialChangedObservable",
  "definedFacingForward",
  "occlusionQueryAlgorithmType",
  "occlusionType",
  "occlusionRetryCount",
  "_occlusionInternalRetryCounter",
  "_isOccluded",
  "_isOcclusionQueryInProgress",
  "_visibility",
  "alphaIndex",
  "isVisible",
  "isPickable",
  "showBoundingBox",
  "showSubMeshesBoundingBox",
  "isBlocker",
  "enablePointerMoveEvents",
  "renderingGroupId",
  "_receiveShadows",
  "renderOutline",
  "outlineColor",
  "outlineWidth",
  "renderOverlay",
  "overlayColor",
  "overlayAlpha",
  "_hasVertexAlpha",
  "_useVertexColors",
  "_computeBonesUsingShaders",
  "_numBoneInfluencers",
  "_applyFog",
  "useOctreeForRenderingSelection",
  "useOctreeForPicking",
  "useOctreeForCollisions",
  "_layerMask",
  "alwaysSelectAsActiveMesh",
  "actionManager",
  "physicsImpostor",
  "_checkCollisions",
  "_collisionMask",
  "_collisionGroup",
  "ellipsoid",
  "ellipsoidOffset",
  "_oldPositionForCollisions",
  "_diffPositionForCollisions",
  "edgesWidth",
  "edgesColor",
  "_collisionsTransformMatrix",
  "_collisionsScalingMatrix",
  "_renderId",
  "_intersectionsInProgress",
  "_unIndexed",
  "_lightSources",
  "_onCollisionPositionChange",
  "onBeforeRenderObservable",
  "onAfterRenderObservable",
  "onBeforeDrawObservable",
  "delayLoadState",
  "instances",
  "_LODLevels",
  "_visibleInstances",
  "_renderIdForInstances",
  "_batchCache",
  "_instancesBufferSize",
  "_originalBuilderSideOrientation",
  "overrideMaterialSideOrientation",
  "_areNormalsFrozen",
  "_source",
  "_geometry",
  "_boundingInfo",
  "subMeshes",
  "_poseMatrix",
  "_preActivateId",
  "_effectiveMaterial"
]
```

Reply:
So we're good!